cli/tools/registry/mod.rs

```diff
@@ -130,11 +125,12 @@ async fn prepare_publish(
     bail!("Invalid package name, use '@<scope_name>/<package_name> format");
   };
 
-  let unfurler = ImportMapUnfurler::new(import_map);
-
-  let (tarball, diagnostics) =
-    tar::create_gzipped_tarball(directory_path, unfurler)
-      .context("Failed to create a tarball")?;
+  let (tarball, diagnostics) = deno_core::unsync::spawn_blocking(move || {
+    let unfurler = ImportMapUnfurler::new(&import_map);
+    tar::create_gzipped_tarball(&dir_path, unfurler)
+      .context("Failed to create a tarball")
+  })
+  .await??;
```

Comment:
Any reason to run it in a separate thread since this code is executed before paralellization happens?